test/api/controllers/ControllerBaseSpec.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.controllers
+
+import api.controllers.ControllerTestRunner.validNino
+import api.mocks.MockIdGenerator
+import api.mocks.services.{MockAuditService, MockEnrolmentsAuthService, MockMtdIdLookupService}
+import api.models.audit.{AuditError, AuditEvent, AuditResponse, GenericAuditDetail}
+import api.models.errors.MtdError
+import play.api.http.{HeaderNames, MimeTypes, Status}
+import play.api.libs.json.JsValue
+import play.api.mvc.{AnyContentAsEmpty, ControllerComponents, Result}
+import play.api.test.Helpers.stubControllerComponents
+import play.api.test.{FakeRequest, ResultExtractors}
+import support.UnitSpec
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.Future
+
+class ControllerBaseSpec
+    extends UnitSpec
+    with Status
+    with MimeTypes
+    with HeaderNames
+    with ResultExtractors
+    with MockAuditService
+    with ControllerSpecHateoasSupport {
+
+  implicit lazy val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest()
+
+  lazy val cc: ControllerComponents = stubControllerComponents()
+
+  lazy val fakeGetRequest: FakeRequest[AnyContentAsEmpty.type] = fakeRequest.withHeaders(
+    HeaderNames.AUTHORIZATION -> "Bearer Token"
+  )
+
+  def fakePostRequest[T](body: T): FakeRequest[T] = fakeRequest.withBody(body)
+}
```

Comment:
Please add val fakeDeleteRequest and def fakePutRequest in this file as shown here https://github.com/hmrc/individuals-state-benefits-api/blob/main/test/api/controllers/ControllerBaseSpec.scala